src/components/dialogs/network-modifications/by-filter-deletion/by-filter-deletion-dialog.tsx

```diff
@@ -33,7 +33,7 @@ import {
 const formSchema = yup
     .object()
     .shape({
-        [TYPE]: yup.string().required(),
+        [TYPE]: yup.mixed<keyof typeof EQUIPMENT_TYPES>().required(),
```

Comment:
Following discussion https://github.com/gridsuite/gridstudy-app/pull/1997#discussion_r1535549043
> why did you set a more generic type for keys?
>> Yeah wasn't too sure of this either there was an incompatibility with the associated `ByFilterDeletionDialog` yup schema which specified `yup.string()` for the field `type`, reverted and adapted yup schema to `yup.mixed<keyof typeof EQUIPMENT_TYPES>()` instead

---

hum... I don't think that is the correct solution, because you lost type check, as[ said in documentation](https://github.com/jquense/yup#mixed):
>Creates a schema that matches all types, or just the ones you configure. Inherits from [Schema](https://github.com/jquense/yup#Schema). Mixed types extends `{}` by default instead of `any` or `unknown`. This is because in TypeScript `{}` means anything that isn't `null` or `undefined` which yup treats distinctly.

Reply:
You can do something like this:
```suggestion
        [TYPE]: yup.oneOf(Object.keys(EQUIPMENT_TYPES)).required(),
```

Reply:
Still need to declare it as `mixed` tho, `yup.oneOf()` as is does not exist. Would `yup.mixed().oneOf(Object.keys(EQUIPMENT_TYPES))` offer better type checking than `yup.mixed<keyof typeof EQUIPMENT_TYPES>()` ?